### PR TITLE
MasterChef Fixes

### LINKED ISSUE #1239: Pool created directly through AMMFactory will stall MasterChef

### LINKED ISSUE #1266: Add events to Rewards Contracts so the Graph can read them
This ensures graph has data for rewards.

### LINKED ISSUE #1271: Add/Remove Transactions not showing up under User Activity
User made a transaction and received a Metamask notification that it was successful; but tx didn't show up on Turbo UI. On Mumbai.

### PR DESCRIPTION
**Pull request by justinbarry**

### CODE REVIEW

**Review by robert-e-davidson3 — APPROVED** [review]
I'm gonna take your word on the tests being correct. The code change looks very straight-forward.

**Review by robert-e-davidson3 — CHANGES_REQUESTED** [review]
Somehow I missed most of the changes. Blocking since I don't know how to undo an approval.

File: packages/smart/contracts/rewards/MasterChef.sol [hunk 1]
```diff
@@ -522,7 +557,21 @@ contract MasterChef is OpenZeppelinOwnable.Ownable {
         RewardPoolLookupInfo memory _rewardPoolLookupInfo =
             rewardPoolLookup[address(_ammFactory)][address(_marketFactory)][_marketId];
 
-        require(_rewardPoolLookupInfo.created, "Reward pool has not been created.");
+        uint256 _pid = _rewardPoolLookupInfo.pid;
```
Comment (robert-e-davidson3) [inline packages/smart/contracts/rewards/MasterChef.sol:565]: This doesn't seem to be read anywhere.

File: packages/smart/contracts/rewards/MasterChef.sol [hunk 2]
```diff
@@ -522,7 +557,21 @@ contract MasterChef is OpenZeppelinOwnable.Ownable {
...
+
+        // If not created should attempt to create it.
+        if (!_rewardPoolLookupInfo.created) {
+            BPool _bPool = _ammFactory.getPool(_marketFactory, _marketId);
+            require(_bPool != BPool(0), "Pool not created.");
```
Comment (robert-e-davidson3) [inline packages/smart/contracts/rewards/MasterChef.sol:565]: Will people be able to make pools with the AMMFactory directly? It looks like if they do then the pool will forever be ineligible for rewards.

**Review by robert-e-davidson3 — DISMISSED** [review]

File: packages/smart/contracts/rewards/MasterChef.sol (hunk 1 above)
Comment (justinbarry) [reply packages/smart/contracts/rewards/MasterChef.sol:565]: Good catch!

**Review by bthaile — APPROVED** [review]
Lgtm

**Pull request merged**